source/material_model/visco_plastic.cc

```diff
@@ -252,7 +252,8 @@ namespace aspect
             out.reaction_terms[i][c] = 0.0;
 
           // Calculate changes in strain invariants and update the reaction terms
-          rheology->strain_rheology.fill_reaction_outputs(in, i, rheology->min_strain_rate, plastic_yielding, out);
+          if (in.requests_property(MaterialProperties::reaction_rates))
```

Comment:
Would this condition still be needed if we added @anne-glerum suggestion in #4865? I suppose this check should save a little bit of computational time regardless, so perhaps it is worth adding both?